### PR TITLE
Fix issue #5519

### PR DESCRIPTION
**Pull request by vegansk**

### CODE REVIEW

File: compiler/lowerings.nim [hunk 1]
```diff
@@ -141,6 +141,7 @@ proc addField*(obj: PType; s: PSym) =
   # because of 'gensym' support, we have to mangle the name with its ID.
   # This is hacky but the clean solution is much more complex than it looks.
   var field = newSym(skField, getIdent(s.name.s & $s.id), s.owner, s.info)
+  incl(field.flags, sfAnon)
```
Comment (Araq) [inline compiler/lowerings.nim]: Look at the code. ``addField`` uses ``s.name.s & $s.id`` as the field name. How come this needs to be added *again* in the codegen then?

Comment (vegansk) [inline compiler/lowerings.nim]: s.id != field.id. s.id is the same for duplicated structure fields, but field.id will differ

**Comment by vegansk** [comment]
@Araq, if I add sfAnon flag to the field symbol, it breaks some tests. For example tparsecombnum.nim fails with:
```
Error: execution of an external compiler program 'gcc -c  -w  -I/home/vega/work/oss/Nim/lib -o tests/misc/nimcache/compiler_parsecomb.o tests/misc/nimcache/compiler_parsecomb.c' failed with exit code: 1

tests/misc/nimcache/compiler_parsecomb.c: In function ‘colonanonymous__KwfZ5vgS4mM7JAIRjswyFg’:
tests/misc/nimcache/compiler_parsecomb.c:3731:43: error: ‘Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w {aka struct Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w}’ has no member named ‘second106916107187’
  map_kGEGluKJNJc0H9cN4ozdbUw((*colonenvP_).second106916107187, T1_, Result);
                                           ^
tests/misc/nimcache/compiler_parsecomb.c: In function ‘star__oItcHiZ0AB9bJn9b9b59atJBmQ’:
tests/misc/nimcache/compiler_parsecomb.c:3742:40: error: ‘Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w {aka struct Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w}’ has no member named ‘second106916107187’
  asgnRefNoCycle((void**) (&(*colonenv_).second106916107187.ClE_0), second.ClE_0);
                                        ^
tests/misc/nimcache/compiler_parsecomb.c:3743:14: error: ‘Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w {aka struct Env_parsecombdotnim81__qEwDi2vrpLBXyHVfT5Os2w}’ has no member named ‘second106916107187’
  (*colonenv_).second106916107187.ClP_0 = second.ClP_0;
              ^
FAILURE
```

File: compiler/lowerings.nim (hunk 1 above)
Comment (Araq) [inline compiler/lowerings.nim]: ``s.id`` should be different.

Comment (vegansk) [inline compiler/lowerings.nim]: Then the bug is in s.id generation.

**Pull request closed without merging**